### PR TITLE
✅ Upgrade metricsql

### PR DESCRIPTION
**Pull request by vegerot**
## Note to reviewer
- Please review this PR [commit-by-commit](https://github.com/jiacai2050/promql-prettier/pull/3/commits) (and do not squash when merging)

# Changes 
✅ Add tests
- Added tests from README
- Refactor a bit of the code to make it more easily testable

⬆️ Upgrade metricsql

- No longer need to maintain fork
- Correctness verified by tests

### CODE REVIEW

**Comment by vegerot** [comment]
If the review would prefer I could make two separate PRs, one for each commit.  And then merge the second PR after the first is merged

**Comment by jiacai2050** [comment]
Thanks for your interests.

Three questions for this PR:
- Why no squash? IMO squash can make the commits history clean. 
- Why replace forked metricsql? I forked it to support UTF-8 chars, does upstream support this out of box?
- I prefer to not use emoji in commit message.

**Comment by vegerot** [comment]
> - Why no squash

- These commits have nothing to do with each other, so it will make the history more searchable.  Don't worry about it, I'll just make separate PRs.  

> Why replace forked metricsql
- I didn't realize it didn't support non-UTF-8 chars.  I just tested what was in the README.  Would you mind giving me an example so I could write a test for it? (or even better, you could contribute that test)

> not use emoji in commit message

done.

**Comment by vegerot** [comment]
Marking as draft until #4 is merged

**Comment by vegerot** [comment]
Circling back to this PR, would you please give me an example of a PromQL expression that metricsql would not parse?

File: util.go [hunk 1]
```diff
@@ -46,11 +46,12 @@ func prettier(expr metricsql.Expr, ident int) []byte {
 		var b bytes.Buffer
 
 		wrapParensWhenNecesary(e.Expr, &b, ident)
-		if len(e.Window) > 0 || len(e.Step) > 0 {
-			b.WriteString(fmt.Sprintf("[%s:%s]", e.Window, e.Step))
+		if (*e.Window).Duration(1) > 0 || e.Step.Duration(1) > 0 {
```
Comment (jiacai2050) [inline util.go:49]: Why this change?

**Comment by jiacai2050** [comment]
```
指标{job="api"}
```
IMO, metricsql can't parse metric name with UTF-8.

I have filed an issue, but seems they have no interests
- https://github.com/VictoriaMetrics/metricsql/issues/1

**Comment by jiacai2050** [comment]
You need to rebase with master, I have added CI in it.

**Comment by jiacai2050** [comment]
Close since no updates.
